### ISSUE TITLE
Long key strings being coerced to unsafe numbers

### ISSUE THREAD

**Issue by greim** [Netflix/falcor-router#176]
I'm returning an array of pathVals from a route. The pathVals contain refs which look like this:

```js
{ $type: 'ref', value: ['assessments', '557430964571080686'] }
```

This in turn feeds back into a different route. That key is being coerced to a number by Falcor router somewhere in between, which is where the trouble starts:

```
'557430964571080686' => 557430964571080700
```

The munged key causes my route to fail. Presumably because the coercion exceeds [`Number.MAX_SAFE_INTEGER`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER). Is there a reason it does the coercion in the first place, and/or a way make Falcor router not coerce strings to numbers?

**Note:** I verified that the route capturing these keys is using `[{keys}]`, not `[{integers}]`.

**Comment by trxcllnt** [comment]
@greim this is captured in #68, and my gut tells me this is an artifact of attempting to collapse ranges when we collapse paths. [See here](https://github.com/Netflix/falcor-path-utils/blob/master/lib/toPaths.js#L135), our [`isNumber` function](https://github.com/Netflix/falcor-path-utils/blob/master/lib/toPaths.js#L209) should probably be smarter (attempt to parse the number, compare length, and compare against `Number.MAX_SAFE_INTEGER`). Only downside is it'll slow down collapse, which is one of the most expensive things in falcor, but it's probably necessary.

**Comment by greim** [comment]
[This PR](https://github.com/Netflix/falcor-path-utils/pull/6/files) seems to fix the issue on my end. There are no unit tests that I could see, but it passed this ad-hoc test:

```js
var assert = require('assert');
assert(!isNumber(null));
assert(!isNumber(true));
assert(!isNumber(false));
assert(!isNumber(""));
assert(!isNumber(Infinity));
assert(!isNumber(-Infinity));
assert(!isNumber('9007199254740992')); // previously, this one failed to pass
assert(!isNumber('-9007199254740992')); // previously, this one failed to pass
assert(!isNumber());
assert(isNumber('9007199254740991'));
assert(isNumber('-9007199254740991'));
assert(isNumber('0'));
assert(isNumber('0x12'));
```

**Comment by trxcllnt** [comment]
@greim also important:

```es6
assert(isNumber('10'));
assert(!isNumber('01'));
```

**Comment by greim** [comment]
Ha, I guess this is kind of a rabbit hole. I added that and threw some unit tests into the PR: https://github.com/Netflix/falcor-path-utils/pull/6/files#diff-eaf71ef7b0b8c9599a8edd0741d594acR155

**Comment by lrowe** [comment]
This should be fixed with the falcor-path-utils 0.5.1 release. Be aware that path collapsing can happen on both client and server.
